packages/core/src/components/button/abstractButton.tsx

```diff
@@ -134,6 +134,16 @@ export abstract class AbstractButton<H extends React.HTMLAttributes<HTMLElement>
         };
     }
 
+    // A disabled element cannot be active, there're are cases where
+    // this situation can happen (NumericInput buttons), hence the need
+    // to check if disabled prop has been passed and change isActive
+    // accordingly
+    public componentDidUpdate() {
```

Comment:
I think this should be implemented inside `static getDerivedStateFromProps` instead:

```ts
    protected static getDerivedStateFromProps(props: IButtonProps, state: IButtonState) {
        if (state.isActive && props.disabled) {
            return {
                isActive: false,
            };
        }

        return null;
    }
```

Reply:
If this is better than componentDidUpdate then no issues, I will change it.

Reply:
It's better because it's a more natural place to manage state updates, it gets called before `render`. Check out the React lifecycle documentation to see what kinds of behavior is recommended in the various methods